File: com.systemincloud.examples.tasks.javatask/src/test/java/tasks/generator/GeneratorPauseResume.java
package tasks.generator;

import com.systemincloud.modeler.tasks.javatask.api.InputPort;
import com.systemincloud.modeler.tasks.javatask.api.JavaTask;
import com.systemincloud.modeler.tasks.javatask.api.OutputPort;
import com.systemincloud.modeler.tasks.javatask.api.annotations.InputPortInfo;
import com.systemincloud.modeler.tasks.javatask.api.annotations.JavaTaskInfo;
import com.systemincloud.modeler.tasks.javatask.api.annotations.OutputPortInfo;
import com.systemincloud.modeler.tasks.javatask.api.data.Control;
import com.systemincloud.modeler.tasks.javatask.api.data.Int32;

@JavaTaskInfo(generator = true)
public class GeneratorPauseResume extends JavaTask {

	@InputPortInfo(name = "In", dataType = Control.class)
	public InputPort in;
	@OutputPortInfo(name = "Out", dataType = Int32.class)
	public OutputPort out;

	private int i = 0;
	
	@Override
	public void generate() {
		sleep(500);
		out.putData(new Int32(i++));
	}

	@Override
	public void execute() {
		if(isPaused()) resume();
		else           pause();
	}

}

File: com.systemincloud.examples.tasks.pythontask/src/test/py/tasks/generator/GeneratorPauseResume.py
from sicpythontask.PythonTaskInfo import PythonTaskInfo
from sicpythontask.PythonTask import PythonTask
from sicpythontask.InputPort import InputPort
from sicpythontask.OutputPort import OutputPort
from sicpythontask.data.Int32 import Int32

@PythonTaskInfo(generator=True)
class GeneratorPauseResume(PythonTask):

    def __init_ports__(self):
        self.in_ = InputPort(name="in")
        self.out = OutputPort(name="out", data_type=Int32)

    def __init__(self):
        self.i = 0

    def generate(self):
        self.sleep(500);
        self.out.put_data(Int32(self.i));
        self.i += 1

    def execute(self, grp):
        if self.paused():
            self.resume();
        else:
            self.pause();
